torchvision/prototype/datasets/_builtin/celeba.py

```diff
@@ -60,15 +59,37 @@ def __iter__(self) -> Iterator[Tuple[str, Dict[str, str]]]:
                 yield line.pop("image_id"), line
 
 
-class CelebA(Dataset):
-    def _make_info(self) -> DatasetInfo:
-        return DatasetInfo(
-            "celeba",
-            homepage="https://mmlab.ie.cuhk.edu.hk/projects/CelebA.html",
-            valid_options=dict(split=("train", "val", "test")),
-        )
+NAME = "celeba"
+
+
+@register_info(NAME)
+def _info() -> Dict[str, Any]:
+    return dict()
+
 
-    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+@register_dataset(NAME)
+class CelebA(Dataset2):
+    """
+    - **homepage**: https://mmlab.ie.cuhk.edu.hk/projects/CelebA.html
+    """
+
+    def __init__(
+        self,
+        root: Union[str, pathlib.Path],
+        *,
+        split: str = "train",
+        skip_integrity_check: bool = False,
+    ) -> None:
+        self._split = self._verify_str_arg(split, "split", ("train", "val", "test"))
+        self._split_id = {
+            "train": "0",
+            "val": "1",
+            "test": "2",
+        }[split]
```

Comment:
Should we just inline that in `_filter_split` where it's used? Also, any reason to use strings here?

Reply:
> Should we just inline that in `_filter_split` where it's used?

:+1:

> Also, any reason to use strings here?

The actual id is read from a csv and thus is a string.